src/Dynamo/README.md

```diff
@@ -0,0 +1,142 @@
+# Dyanmo
+
+Dynamo is code generator that uses C# combinators to write source code. The target language can be any language you wish. At present, it is built to support C# and Swift which are requirements for Binding Tools for Swift.
+
+Dynamo was meant to be a step up from using `Console.WriteLine` and as such doesn't care about the order in which the code is being generated. This is important in Binding Tools for Swift in that it allows locality of responsibility. Consider the process of writing a binding for an enum from Swift. If we start with the following Swift:
```

Comment:
It would be helpful to include the requirements for when it is recommended to use: as non-linear projections, Swift code, nested blocks, etc.